From 24b8b4e5df644ba77c673c41b90cc48c4358aaa0 Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fvogt@suse.de>
Date: Thu, 2 Dec 2021 15:50:08 +0100
Subject: [PATCH] Make H.264 codec optional during runtime

It's possible that FreeRDP was built against FFmpeg, but it doesn't support
H.264. In that case, just continue without H.264 support instead of failing
hard before even trying to connect.

This is especially useful for Linux distributions which can't ship H.264
support in FFmpeg out of the box (patent issues), but allow enabling H.264
later by installing a version of FFmpeg which has it enabled.
---
 libfreerdp/core/codecs.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

Index: FreeRDP-2.4.1/libfreerdp/core/codecs.c
===================================================================
--- FreeRDP-2.4.1.orig/libfreerdp/core/codecs.c
+++ FreeRDP-2.4.1/libfreerdp/core/codecs.c
@@ -106,10 +106,7 @@ BOOL freerdp_client_codecs_prepare(rdpCo
 
 		if (!(codecs->h264 = h264_context_new(FALSE)))
 		{
-			WLog_ERR(TAG, "Failed to create h264 codec context");
-#ifndef WITH_OPENH264_LOADING
-			return FALSE;
-#endif
+			WLog_WARN(TAG, "Failed to create h264 codec context");
 		}
 	}
 #endif
